### ISSUE TITLE
BigEndian support

### ISSUE THREAD

**Issue by UkooLabs** [jkuhlmann/cgltf#150]
would it be possible to add support for BigEndian

**Comment by jkuhlmann** [comment]
For what would the be applicable? Are you having issues with endianess? Where does it break?

**Comment by UkooLabs** [comment]
@jkuhlmann in my case Xbox 360. I have made some modifications which has made it work. But maybe a better more complete way to do it that also supports writing as only done some read side.

I'm not sure what other platforms I'll be using that will also need big endian yet.

**Comment by jkuhlmann** [comment]
Yeah, I think there aren't too many platforms left that default to big endian nowadays. The Xbox 360 seems hardly relevant when the Series X is out. :wink: 
Anyway, if you could share some sample code of what you fixed and how you did it, that might help me make a better decision on what we can do and if we should do something.

**Comment by UkooLabs** [comment]
[Changes.zip](https://github.com/jkuhlmann/cgltf/files/7165988/Changes.zip)

I have attached a zip showing my changes. You will see int.h which add various int types not available for XBOX Original. Then the endian helper which checks for endianness and can convert values.

Since im not 100% familular with clgtf I may of missed conversion of endianess, if you see any please let me know but loading models on a Xbox 360 seems to work fine and render.

**Comment by cblc** [comment]
While big endian systems are not common nowadays, however IMHO I think this library should be endianness-independent, because when you see C code, you expect to build everywhere. All my source code still builds on PowerPC Macs and on IRIX MIPS, not that I still use that hardware nowadays, but I consider being able to still build on them as a measure of code quality.
